ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+
+    //hMiraMonLayer->Version = MM_VECTOR_LAYER_LAST_VERSION;
+    hMiraMonLayer->nMemoryRatio = nMMMemoryRatio;
+
+    hMiraMonLayer->ReadOrWrite = ReadOrWrite;
+    hMiraMonLayer->MMMap = MMMap;
+
+    // Don't free in destructor
+    hMiraMonLayer->pLayerDB = pLayerDB;
+
+    // Opening mode
+    strcpy(hMiraMonLayer->pszFlags, "wb+");
+
+    if (LayerVersion == MM_UNKNOWN_VERSION)
+    {
+        MMCPLError(CE_Failure, CPLE_NotSupported,
+                   "Unknown version in MiraMon driver.");
+        return 1;
+    }
+    if (LayerVersion == MM_LAST_VERSION)
+    {
+        MMSet1_1Version(&hMiraMonLayer->TopHeader);
+        hMiraMonLayer->nHeaderDiskSize = MM_HEADER_SIZE_64_BITS;
+        hMiraMonLayer->LayerVersion = MM_64BITS_VERSION;
+    }
+    else if (LayerVersion == MM_32BITS_VERSION)
+    {
+        MMSet1_1Version(&hMiraMonLayer->TopHeader);
+        hMiraMonLayer->nHeaderDiskSize = MM_HEADER_SIZE_32_BITS;
+        hMiraMonLayer->LayerVersion = MM_32BITS_VERSION;
+    }
+    else
+    {
+        MMSet2_0Version(&hMiraMonLayer->TopHeader);
+        hMiraMonLayer->nHeaderDiskSize = MM_HEADER_SIZE_64_BITS;
+        hMiraMonLayer->LayerVersion = MM_64BITS_VERSION;
+    }
+
+    hMiraMonLayer->pszSrcLayerName = strdup_function(pzFileName);
+    hMiraMonLayer->szLayerTitle =
+        strdup_function(get_filename_function(pzFileName));
+
+    if (!hMiraMonLayer->bIsBeenInit &&
+        hMiraMonLayer->eLT != MM_LayerType_Unknown)
+    {
+        if (MMInitLayerByType(hMiraMonLayer))
+        {
+            // Error specified inside the function
+            return 1;
+        }
+        hMiraMonLayer->bIsBeenInit = 1;
+    }
+
+    // If more nNumStringToOperate is needed, it'll be increased.
+    hMiraMonLayer->nNumStringToOperate = 0;
+    if (MMResizeStringToOperateIfNeeded(hMiraMonLayer, 500))
+    {
+        MMCPLError(CE_Failure, CPLE_OutOfMemory,
+                   "Memory error in MiraMon "
+                   "driver (MMInitLayer())");
+        return 1;
+    }
+
+    hMiraMonLayer->nMMLanguage = nMMLanguage;
+
+    if (nMMRecode == MM_RECODE_UTF8)
+        hMiraMonLayer->nCharSet = MM_JOC_CARAC_UTF8_DBF;
+    else  //if(nMMRecode==MM_RECODE_ANSI)
+        hMiraMonLayer->nCharSet = MM_JOC_CARAC_ANSI_DBASE;
+    return 0;
+}
+
+/* -------------------------------------------------------------------- */
+/*      Layer Functions: Closing MiraMon layers                         */
+/* -------------------------------------------------------------------- */
+static int MMClose3DSectionLayer(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                                 MM_INTERNAL_FID nElements, FILE_TYPE *pF,
+                                 FILE_TYPE *pF3d, const char *pszF3d,
+                                 struct MM_ZSection *pZSection,
+                                 MM_FILE_OFFSET FinalOffset)
+{
+    if (!hMiraMonLayer)
+        return 1;
+
+    // Flushing if there is something to flush on the disk
```

Comment:
This comment doesn't seem to be related with the null pointer checks just below

Reply:
Oh. Sure.

// Avoid closing when it has no sense. But it's not an error.
    // Just return elegantly.